setup/src/Magento/Setup/Mvc/Bootstrap/InitParamListener.php

```diff
@@ -138,25 +136,26 @@ private function extractInitParameters(Application $application)
                 $result[$initKey] = $_SERVER[$initKey];
             }
         }
-        $result = array_replace_recursive($result, $this->extractFromCli($application->getRequest()));
-        return $result;
+
+        if (!isset($result['argv'])) {
```

Comment:
Please add the following check in order to prevent type error if it's not array:
```suggestion
        if (!isset($result['argv']) || !is_array($result['argv'])) {
```
Also would be great to cover this case with a unit test

Reply:
Thank you, will do

Reply:
Done